src/tools/miri/src/concurrency/thread.rs

```diff
@@ -1045,21 +1046,22 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
     /// Run the core interpreter loop. Returns only when an interrupt occurs (an error or program
     /// termination).
     fn run_threads(&mut self) -> InterpResult<'tcx, !> {
-        static SIGNALED: AtomicBool = AtomicBool::new(false);
+        // In normal rustc, rustc_driver::main installs this handler. But we don't use that
+        // function, see src/bin/miri.rs.
```

Comment:
If this is a "thing the driver would do", then it probably makes more sense to install this in `miri.rs` than in the library.

Reply:
Probably. I factored the installation into a `rustc_driver` function that we call in `miri.rs`.